src/main/java/worldofzuul/Rooms/BuildArea.java

```diff
@@ -21,7 +21,7 @@ public BuildArea() {
      */
     public void addEnergySource(EnergySource e) {
         // Clone the energysource, to prevent mutation when setting a new position
-        var energySourceClone = new EnergySource(e.getName(), e.getDescription(), e.getSize(), e.getPrice(), e.getEmission(), e.getOutput(), e.getCapacity(), e.getTextureURL());
+        var energySourceClone = new EnergySource(e.getName(), e.getDescription(), e.getSize(), e.getPrice(), e.getEmission(), e.getOutput(), e.getCapacity(), e.getTextureURL(), e.getWidth(), e.getHeight());
```

Comment:
```suggestion
        EnergySource energySourceClone = new EnergySource(e.getName(), e.getDescription(), e.getSize(), e.getPrice(), e.getEmission(), e.getOutput(), e.getCapacity(), e.getTextureURL(), e.getWidth(), e.getHeight());
```

Why you use VAR my guy, seems a little cursed.
otherwise, it seems smart.